posts/Martin/2015-07-13-introduction-to-hibernate-search-genericjpa.adoc

```diff
@@ -0,0 +1,460 @@
+org.hibernate.search.genericjpa.db.events.triggers.MySQLTriggerSQLStringSource
+ 
+# we have a searchfactory that is supposed 
+#to get it's updates from SQL
+org.hibernate.search.genericjpa.searchfactory.type=sql
+ 
+# how many updates should be processed at once?
+# this number is quite conservative
+org.hibernate.search.genericjpa.searchfactory.batchSizeForUpdates=10
+ 
+# what delay do we want between checking for updates?
+org.hibernate.search.genericjpa.searchfactory.updateDelay=25
+ 
+# we are in a Java SE environment in this example.
+# if you specify 'true', you will have to add an extra property
+# to lookup the JTA mechanism. This is however not needed
+# if you are managing your Transactions by yourself
+# org.hibernate.search.genericjpa.searchfactory.transactionManagerProvider.jndi=\
+# java:jboss/TransactionManager
+org.hibernate.search.genericjpa.searchfactory.useJTATransactions=false
+ 
+# for this example we store our directory in ram.
+hibernate.search.default.directory_provider=ram
+ 
+# we could add extra properties as specified in the Hibernate Search 
+# documentation (http://docs.jboss.org/hibernate/search/5.3/reference/en-US/html_single/#_configuration) next
+----
+
+Now we could start using hibernate search for queries.
+
+[source,java]
+----
+EntityManager em = ...;
+em.getTransaction().begin();
+Book book = new Book();
+book.setName( "The Hobbit" );
+book.setAuthor( "J.R.R. Tolkien");
+em.persist( book );
+em.getTransaction().commit();
+ 
+FullTextEntityManager fem = searchController.getFullTextEntityManager( em );
+ 
+QueryBuilder qb = fem.getSearchFactory()
+  .buildQueryBuilder().forEntity(Book.class).get();
+org.apache.lucene.search.Query query = qb
+  .keyword()
+  .onFields("name")
+  .matching("The Hobbit")
+  .createQuery();
+ 
+List books = fem.createFullTextQuery(
+  query, Book.class )
+    .getResultList();
+----
+
+But wait, why didn’t we receive any Books even though our query was right?
+Well, that’s because we forgot to add some additional classes to our persistence.xml.
+In order to keep the index up-to-date we have to specify special entities that can hold
+the information about changes in the database. These will be queried by our
+Hibernate Search GenericJPA engine and then the index is updated accordingly.
+For our example, we need this special entity:
+
+[source,java]
+----
+package org.hibernate.search.genericjpa.test.entities;
+ 
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+ 
+import org.hibernate.search.genericjpa.annotations.Event;
+import org.hibernate.search.genericjpa.annotations.IdFor;
+import org.hibernate.search.genericjpa.annotations.Updates;
+ 
+@Entity(name = "BookUpdates")
+// what is the original table name? and what table is this entity saved to
+// this is kinda duplicated, but needed
+@Updates(originalTableName = "Book", tableName = "BookUpdates")
+@Table(name = "BookUpdates")
```

Comment:
Also, this code isn't that easily written... I have to map every table in the whole index tree including the collection tables. I just thought that for a first version this would be too much work for me during this summer of code.